pkg/vault/contracts/RouterCommon.sol

```diff
@@ -83,6 +87,15 @@ abstract contract RouterCommon is IRouterCommon, VaultGuard {
         _discardSenderIfRequired(isExternalSender);
     }
 
+    /// @notice Locks the return of excess ETH to the sender until the end of the function.
+    /// @dev This modifier must be used after `saveSender` to ensure that the sender is saved.
+    modifier useMulticall() {
+        _isReturnEthLockedSlot().tstore(true);
+        _;
+        _isReturnEthLockedSlot().tstore(false);
+        _returnEth(_getSenderSlot().tload());
+    }
```

Comment:
Seems error-prone to require two modifiers. Can we just combine them? Repeating "saveSender" in this one's name so that people don't try to use both anyway.

```suggestion
    /**
     * @notice Locks the return of excess ETH to the sender until the end of the function.
     * @dev This also encompasses the `saveSender` functionality.
     */
    modifier saveSenderAndManageEth() {
        bool isExternalSender = _saveSender();

        // Lock the return of ETH during execution
        _isReturnEthLockedSlot().tstore(true);
        _;
        _isReturnEthLockedSlot().tstore(false);

        _returnEth(_getSenderSlot().tload());
        _discardSenderIfRequired(isExternalSender);
    }
```